primitives/src/vid/advz/namespace.rs

```diff
@@ -0,0 +1,355 @@
+// Copyright (c) 2023 Espresso Systems (espressosys.com)
+// This file is part of the Jellyfish library.
+
+// You should have received a copy of the MIT License
+// along with the Jellyfish library. If not, see <https://mit-license.org/>.
+
+//! Implementation of [`Namespacer`] for `Advz`.
+
+use ark_poly::EvaluationDomain;
+use jf_utils::{bytes_to_field, compile_time_checks};
+
+use super::{
+    AffineRepr, Debug, DenseUVPolynomial, Digest, DynDigest, GenericAdvz, MerkleTreeScheme,
+    PolynomialCommitmentScheme, PrimeField, UnivariatePCS, Vec, Write,
+};
+use crate::{
+    alloc::string::ToString,
+    vid::{namespace::Namespacer, vid, VidError},
+};
+use ark_std::format;
+
+impl<P, T, H, V> Namespacer for GenericAdvz<P, T, H, V>
+where
+    // TODO ugly trait bounds https://github.com/EspressoSystems/jellyfish/issues/253
+    P: UnivariatePCS<Point = <P as PolynomialCommitmentScheme>::Evaluation>,
+    P::Evaluation: PrimeField,
+    P::Polynomial: DenseUVPolynomial<P::Evaluation>,
+    P::Commitment: From<T> + AsRef<T>,
+    T: AffineRepr<ScalarField = P::Evaluation>,
+    H: Digest + DynDigest + Default + Clone + Write,
+    V: MerkleTreeScheme<Element = Vec<P::Evaluation>>,
+    V::MembershipProof: Sync + Debug,
+    V::Index: From<u64>,
+{
+    // TODO should be P::Proof, not Vec<P::Proof>
+    // https://github.com/EspressoSystems/jellyfish/issues/387
+    type DataProof = Vec<P::Proof>;
+
+    fn data_proof(
```

Comment:
In the namespace application, does the prover need to generate these proofs online? Can it just batchly preprocess all of the evaluation proofs offline, and in the online phase, whenever a client asks, it just forwards proofs without doing any computations?
cc @ggutoski @jbearer

Reply:
Yes, it could. I suspect servers will opt to pre-process everything if they get sufficient traffic. Seems to me that the the priority is to optimize data on the wire, so KZG batch proofs should be quite helpful here.

Reply:
#397